File: server-purs/src/Control/Monad/Aff/FileSystem.js
exports._read = function _read(path) {
  return function (fail, succeed) {
    var fs = require('fs')
    fs.readFile(path, function (error, data) {
      if (error) fail(error)
      else succeed(data.toString())
    })
  }
}

exports._write = function _write(unit, path, data) {
  return function (fail, succeed) {
    var fs = require('fs')
    fs.writeFile(path, data, function (error) {
      if (error) fail(error)
      else succeed(unit)
    })
  }
}

exports._exists = function _exists(path) {
  return function (fail, succeed) {
    var fs = require('fs')
    fs.exists(path, succeed)
  }
}

exports._createDirectory = function _createDirectory(unit, path) {
  return function (fail, succeed) {
    var mkdirp = require('mkdirp')
    mkdirp(path, function (error) {
      if (error) fail(error)
      else succeed(unit)
    })
  }
}

File: lib/ellie/system/file_system.ex
defmodule Ellie.System.FileSystem do
  def create_temporary_directory() do
    case System.tmp_dir() do
      nil -> :error
      path -> {:ok, path}
    end
  end

  def write(path, data) do
    File.write(path, data)
  end

  def read(path) do
    File.read(path)
  end

  def remove_directory(path) do
    File.rm_rf(path)
  end
end
